### PR TITLE
Doxygen improvements for sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90

### PR DESCRIPTION
**Pull request by lgannoaa**
Part of #366

### CODE REVIEW

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 [hunk 1]
```diff
@@ -42,69 +44,72 @@ module pmat
 
 contains
 
-!=============================================================================
+!> Swap vectors
+!!
+!! @param d vector
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:49]: Always use @param[in] @param[out] or @param[inout]

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 [hunk 2]
```diff
@@ -42,69 +44,72 @@ module pmat
...
 real(sp),    intent(inout) :: d(:), e(:)
 real(sp)                   :: tv(size(d))
-!=============================================================================
 tv = d; d = e; e = tv
 end subroutine sswpvv
-!=============================================================================
+
+!> dswpvv routine
+!! @author R. J. Purser
 subroutine dswpvv(d,e)!                                                [swpvv]
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:60]: Always list all params, in order.

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 [hunk 3]
```diff
@@ -42,69 +44,72 @@ module pmat
 
 contains
 
-!=============================================================================
+!> Swap vectors
+!!
+!! @param d vector
+!! @param e vector
+!! @author R. J. Purser
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:51]: Always put one empty documentation line before listing parameters.

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 [hunk 4]
```diff
@@ -42,69 +44,72 @@ module pmat
...
 subroutine sinvmt(a)!                                                    [inv]
-!=============================================================================
 real(sp),dimension(:,:),intent(INOUT):: a
 logical                              :: ff
 call sinvmtf(a,ff)
 if(ff)stop 'In sinvmt; Unable to invert matrix'
 end subroutine sinvmt
-!=============================================================================
+
+!> dinvmt routine
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:83]: Always capitalize and punctuate, especially first sentence. This has been mentioned many times.

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 [hunk 5]
```diff
@@ -113,36 +118,35 @@ subroutine sinvmtf(a,ff)!                                                [inv]
    print '(" In sinvmtf; failed call to sldumf")'
    return
 endif
-
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:116]: Why did you remove these blank lines?

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 [hunk 6]
```diff
@@ -404,57 +406,55 @@ subroutine iinvf(imat,ff)!                                               [inv]
 endif
 end subroutine iinvf
 
-!=============================================================================
+!> routine sldum
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:409]: If all we know is the routine name, replace "routine XXX" with three question marks ("???"). We will ask the original author to come fill those in.

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 [hunk 7]
```diff
@@ -630,25 +630,21 @@ subroutine cldumf(a,ipiv,d,ff)!                                         [ldum]
 enddo
 end subroutine cldumf
 
-!=============================================================================
+!> @brief SUBROUTINE UDLMM.
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:633]: Do not use brief tags with subroutines functions.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 (hunk 1 above)
Comment (lgannoaa) [reply sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:49]: done

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 (hunk 3 above)
Comment (lgannoaa) [reply sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:51]: done

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 (hunk 2 above)
Comment (lgannoaa) [reply sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:60]: Which param is not in order?

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 (hunk 4 above)
Comment (lgannoaa) [reply sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:83]: We normally direct copy the comment from original author's code remark. Looked at the record, can't see specified many times.

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 (hunk 5 above)
Comment (lgannoaa) [reply sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:116]: The blank line here is inconsistent for the coding style. I can add it back if you want.

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 (hunk 6 above)
Comment (lgannoaa) [reply sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:409]: done

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 (hunk 7 above)
Comment (lgannoaa) [reply sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:633]: done

**Comment by lgannoaa** [comment]
Just updated. Please check it again

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 (hunk 2 above)
Comment (edwardhartnett) [reply sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:63]: In other cases, in your doxygen changes.

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 (hunk 5 above)
Comment (edwardhartnett) [reply sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:116]: I am in favor of consistent style, but it will not be achieved until we agree what it is, and the provide some automatic formatting. However, that's not what we are doing now. At the moment, doxygen changes only in these doxygen PRs.

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 [hunk 8]
```diff
@@ -113,36 +133,35 @@ subroutine sinvmtf(a,ff)!                                                [inv]
...
    do i=1,j; a(i,j)=a(i,j)+dot_product(a(jp:m,j),a(i,jp:m)); enddo
    do i=jp,m; a(i,j)=dot_product(a(i:m,j),a(i,i:m));         enddo
 enddo
-
 !  Permute columns according to ipiv
 do j=m-1,1,-1; l=ipiv(j); call sswpvv(a(:,j),a(:,l)); enddo
 end subroutine sinvmtf
-!=============================================================================
+
+!> Routine dinvmtf to Invert and Permute
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:154]: Punctuate all sentences.

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 [hunk 9]
```diff
@@ -276,15 +292,15 @@ subroutine dlinmmtf(a,b,ff)!                                             [inv]
 endif
 call dudlmm(a,b,ipiv)
 end subroutine dlinmmtf
-!=============================================================================
+
+!> ???
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:296]: Missing params.

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 [hunk 10]
```diff
@@ -297,42 +313,44 @@ subroutine clinmmtf(a,b,ff)!                                             [inv]
 call cudlmm(a,b,ipiv)
 end subroutine clinmmtf
 
-!=============================================================================
+!> ???
+!! @author R. J. Purser
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:317]: Missing params.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]
Still some functions without params listed, and some other problems. See comments.

**Review by edwardhartnett — APPROVED** [review]

**Comment by edwardhartnett** [comment]
I will merge these changes next...
